src/tictactoe/game.clj

```diff
@@ -0,0 +1,62 @@
+(ns tictactoe.game
+  (:require [tictactoe.user-input-validation :as validation])
+  (:require [tictactoe.user-input :as user-input])
+  (:require [tictactoe.board :as board])
+  (:require [tictactoe.ui :as ui])
+  (:require [tictactoe.detect-board-state :as detect-board-state])
+  (:require [tictactoe.board-translators :as board-translators]))
+
+(defn switch-player [player]
+  (* -1 player))
+
+(defn get-move [game]
+  (let [{ui-board :ui-board board
+         :board ui->board
+         :ui->board} game]
+    (loop [proposed-move (user-input/get-user-move)]
+      (let [validation-results
+            (->>
+              [proposed-move nil]
+              (validation/valid-or-error #(validation/valid-ui-choice? % ui-board))
+              (validation/valid-or-error #(validation/open-square? (ui->board %) board)))]
+          (let [[move error] validation-results]
+            (if (nil? move)
+              (do
+                (ui/render-msg error)
+                (recur (user-input/get-user-move)))
+              (assoc game :move move)))))))
```

Comment:
https://github.com/mkrump/tic-tac-toe-clojure/pull/6/commits/1f588cced4b24a63598e252af47469e7fe337e79 Separated out validation in the Players branch.